docker-entrypoint.sh

```diff
@@ -36,6 +36,13 @@ if [ "$1" = 'postgres' ] && [ "$(id -u)" = '0' ]; then
 	mkdir -p /var/run/postgresql
 	chown -R postgres /var/run/postgresql
 	chmod g+s /var/run/postgresql
+    # Create the transaction log directory before initdb is run (below) so the directory is owned by the correct user
+    if [ "$POSTGRES_XLOG_DIR" ]; then
+        mkdir -p "$POSTGRES_XLOG_DIR"
+        chmod 700 "$POSTGRES_XLOG_DIR"
+        chown -R postgres "$POSTGRES_XLOG_DIR"
+        POSTGRES_INITDB_ARGS="$POSTGRES_INITDB_ARGS --xlogdir $POSTGRES_XLOG_DIR"
+    fi
```

Comment:
Thanks for your patience and good-nature, @DanielDent :disappointed: :heart:

I'm not _100%_ sold, but I concede that we probably need something for this.  Here are my thoughts on the implementation:

- I'd like a blank space above the comment (so that the `/var/run/postgresql` lines are separated from these)
- I'd rather name the variable `POSTGRES_INITDB_XLOGDIR` so it makes it more clear what it interacts with (and maps directly to the option it enables)
- the `POSTGRES_INITDB_ARGS="$POSTGRES_INITDB_ARGS ..."` line should start with `export`,
  ie `export POSTGRES_INITDB_ARGS="$POSTGRES_INITDB_ARGS ..."`
- the four-space indent you've got should be tabs (matching the rest of the file) :+1:

Happy to take over for any of this if you'd prefer, but want to give you first dibs (and a chance to converse if what I've proposed doesn't make sense or isn't accurate :smile:) :heart: